alica_engine/include/engine/model/PlaceholderMapping.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "AlicaElement.h"
+#include <unordered_map>
+
+namespace alica
+{
+
+class PlaceholderMappingFactory;
+class AbstractPlan;
+
+class PlaceholderMapping : public AlicaElement
```

Comment:
This isn't really an alica element right, since it is not created in plan designer?  It is more just a config on par with Alica.yaml file.  In particular I think it may not have id/name

Reply:
Another thing to note is in the current implementation we can map either a behaviour or a plan to a placeholder. This is pretty apparent I guess, but just putting it here so that there is no confusion

Reply:
Yeah this is intentional (though really now that plans have run() method etc, behaviours are basically just plans with a single entrypoint/state)

Reply:
I spent quite a bit of time on this but I think from an implementation perspective it is easiest to treat the mapping as an alica element. However, I did change how the mapping is specified as you suggested. Now the engine takes the mapping json directly instead of a file name. Additionally there is no need to specify id, name & comment for the mapping. So basically the mapping is treated as an alica element only within the engine as an implementation convenience